Update Security docs and properties files (#4309)
quarkusio/quarkus: docs/src/main/asciidoc/security-guide.adoc

and pull requests should be submitted there:
https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
////
= Quarkus - Security Guide

include::./attributes.adoc[]

> Quarkus comes with build in securuty to allow for RBAC based on thecommon security annotations `@RolesAllowed`,

Reviewer: ```suggestion
Quarkus comes with build in security to allow for RBAC based on the common security annotations `@RolesAllowed`,
```

<4> This call to obtain the user principal will return null if the caller is unauthenticated, non-null if the caller is authenticated.
<5> The `/subject/denied` endpoint disallows any access regardless of whether the call is authenticated by specifying the `@DenyAll` annotation.

## Security implementations

Quarkus comes with several different Security extensions that provide different functionality. This functionality is
very much a work in progress, so this list will be expanded over the coming weeks.

.Security Extensions
|===
|Extension |Description

|link:elytron-properties-guide.html[quarkus-elytron-security-properties-file]
|Provides support for simples properties files that can be used for testing security. This supports both embedding user info in `application.properties` and standalone properties files.

> |link:oauth2-guide.html[quarkus-elytron-security-oath2]

Reviewer: ```suggestion
|link:oauth2-guide.html[quarkus-elytron-security-oauth2]
```

> |Provides support for OAth2 flows using Elytron. This extension will likely be deprecated soon and replaced by a reactive Vert.x version.

Reviewer: ```suggestion
|Provides support for OAuth2 flows using Elytron. This extension will likely be deprecated soon and replaced by a reactive Vert.x version.
```

|link:jwt-guide.html[quarkus-smallrye-jwt]

> |A Microprofile JWT implementation that provides support for authenticating use Json Web Tokens. This also allows you to inject the token and claims into the application as per the MP JWT spec.

Reviewer: ```suggestion
|A Microprofile JWT implementation that provides support for authenticating using Json Web Tokens. This also allows you to inject the token and claims into the application as per the MP JWT spec.
```

> |link:keycloak-guide.html[quarkus-vertx-keycloak]

Reviewer: why did we rename it `vertx-keycloak`? It doesn't work at all in a servlet environment?

Author: Because our naming format is productname-feature. In this case vertx is providing Keycloak, so according to our standard naming rules this should be the name. I am happy to change it to just keycloak, but it might even need to be something more generic (as it may just be the open id connect implementation)

Reviewer: Can we have a discussion about the naming on the mailing list? In respect to the `elytron` thing and `vertx-keycloak`? Let's not block this PR for that.
